stakedb/stakedb.go

```diff
@@ -146,7 +251,8 @@ func NewStakeDatabase(client *rpcclient.Client, params *chaincfg.Params,
 
 	stakeDBPath := filepath.Join(dataDir, DefaultStakeDbName)
 	if err = sDB.Open(stakeDBPath); err != nil {
-		return nil, err
+		_ = poolDB.Close()
```

Comment:
If you are not capture the error here `_ = poolDB.Close()` looks like something is missing. I think it would look neater just as `poolDB.Close()`

Reply:
It sort of says "I've not forgotten the error, I just don't care."  So I've added a comment on each one to document this.  It also shuts up linters that look for unhandled errors.